src/cmd/cmdmsgs.rb.in

```diff
@@ -14,148 +14,52 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# We use 'dl' for Ruby <= 1.9.x and 'fiddle' for Ruby >= 2.0.x
-if RUBY_VERSION.split('.')[0] < '2'
-  require 'dl'
-  require 'dl/import'
-  include DL
-else
-  require 'fiddle'
-  require 'fiddle/import'
-  include Fiddle
-end
-
-require 'optparse'
+require 'open3'
 
 # Constants.
-LIBRARY_NAME = '@library_location@'
 LIBRARY_VERSION = '@PROJECT_VERSION_FULL@'
-COMMON_OPTIONS =
-               "  -h [ --help ]              Print this help message.\n"\
-               "  --force-version <VERSION>  Use a specific library version.\n"\
-               '  --versions                 Show the available versions.'
-COMMANDS = {  'msg' =>
-                  "Print information about Gazebo messages.\n\n" +
-                  "  gz msg [options]\n\n" +
-                  "Options:\n\n" +
-                  "  -i [ --info ] arg          " +
-                  "Get info about the specified message type.\n" +
-                  "  -l [ --list ]              List all message types.\n" +
-                  COMMON_OPTIONS
+COMMANDS = {
+  "plugin" => "@plugin_exe_location@",
 }
 
 #
-# Class for the Gazebo msgs command line tools.
+# Class for the Gazebo @GZ_DESIGNATION@ command line tools.
 #
 class Cmd
-  #
-  # Return a structure describing the options.
-  #
-  def parse(args)
-    options = {}
-
-    usage = COMMANDS[args[0]]
-
-    # Read the command line arguments.
-    opt_parser = OptionParser.new do |opts|
-      opts.banner = usage
-
-      opts.on('-h', '--help', 'Print this help message') do
-        puts usage
-        exit(0)
-      end
-
-      opts.on('-l', '--list', 'Print all message types') do |l|
-        options['list'] = l
-      end
-
-      opts.on('-i msg', '--info', String,
-              'Print information about a message') do |t|
-        options['info'] = t
-      end
-    end
-    begin
-      opt_parser.parse!(args)
-    rescue
-      puts usage
-      exit(-1)
-    end
-
-    # Check that there is at least one command and there is a plugin that knows
-    # how to handle it.
-    if ARGV.empty? || !COMMANDS.key?(ARGV[0]) ||
-       options.empty?
-      puts usage
-      exit(-1)
-    end
-
-    options['command'] = ARGV[0]
-
-    options
-  end  # parse()
-
   def execute(args)
-    options = parse(args)
-
-    # puts 'Parsed:'
-    # puts options
+    command = args[0]
+    exe_name = COMMANDS[command]
 
-    # Read the plugin that handles the command.
-    if LIBRARY_NAME[0] == '/'
+    if exe_name[0] == '/'
```

Comment:
I don't think this works on windows. I recommend using `pathname` as
https://github.com/gazebosim/gz-sim/blob/001dd9b829eb8e8b4465d87c6c70ccf4ee2e6b6a/src/cmd/cmdsim.rb.in#L355